linode/manager
# upcoming: [DI: 21998] - Added Service Type, Engine Option and Region Select component to Create Alert form

## Description 📝
- Added Service, Engine Option, Region Select components to the Create Alert form under General Information.
- Added Unit tests for all three components
## Changes  🔄
- Added Autocomplete components for service type, engine option, region selection for the Create Alert form
- Added Unit tests for the components
- Modified the Alert Create `POST` query to include service_type as a parameter of the request
- Minor changes to validation schema to improve the error messages
- Minor changes to types and interfaces 

## Target release date 🗓️

Please specify a release date (and environment, if applicable) to guarantee timely review of this PR. If exact date is not known, please approximate and update it as needed.

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-11-13 at 19 17 38](https://github.com/user-attachments/assets/b3a89732-53ab-4cb3-b9a1-e7a8edc46704) | ![Screenshot 2024-11-19 at 21 38 14](https://github.com/user-attachments/assets/5c482b69-527e-4848-8cf9-fbcc2c5531ec) ![Screenshot 2024-11-19 at 21 38 24](https://github.com/user-attachments/assets/ca92b045-925a-4ec0-bc12-65821086f73a)|

## How to test 🧪

### Prerequisites

(How to setup test environment)

- The tabs are controlled by a featureFlag called aclpAlerting, the flag has been currently disabled. For testing enable the definitions part of the aclpAlerting flag to be true.
- The API endpoints are not live yet, so use Legacy MSW Handlers for the mock responses.

### Verification steps

(How to verify changes)
- In Monitor (once the prerequisites are followed) , Alerts tab should be visible next to Dashboards.
- Under that,  Definition tab and a Create button should be visible.
- On clicking Create, the Form Page should be visible.
- The Engine Option component is designed to be visible only when the Service type is Databases, so selecting Databases in Service renders the Engine Option component
## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅
